firmware/projects/MyBlink/platforms/cli/bindings.cc

```diff
@@ -0,0 +1,42 @@
+#include <stdio.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <iostream>
+#include <signal.h>
+#include "../../bindings.h"
+#include "mcal/cli/periph/gpio.h"
+#include "shared/periph/gpio.h"
+
+namespace mcal {
+
+cli::periph::DigitalOutput indicator{"Indicator"};
+
+}  // namespace mcal
+
+//app level function
+bool bindings::toggle = false;
+
+//app level function that takes care of the interupt handler
+void AppLevelFunction(){
```

Comment:
The point of separating the app and platform layers is to allow our app level functions to be defined once in `main.cc` and work on any platform. Here, you have defined AppLevelFunction in `cli/bindings.cc`, which means you would have to re-implement it in `stm32f767/bindings.cc` to compile on that platform.

This function should be moved to `main`, then you can `extern` reference it in bindings to attach it to the signal handler